operators/azure/openai_invokers.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "openai_invokers.hpp"
+
+namespace ort_extensions {
+
+OpenAIAudioToTextInvoker::OpenAIAudioToTextInvoker(const OrtApi& api, const OrtKernelInfo& info)
+    : CurlInvoker(api, info) {
+  audio_format_ = TryToGetAttributeWithDefault<std::string>(kAudioFormat, "");
+
+  const auto& property_names = PropertyNames();
+
+  const auto& find_optional_input = [&property_names](const std::string& property_name) {
+    std::optional<size_t> result;
+    auto optional_input = std::find_if(property_names.begin(), property_names.end(),
+                                       [&property_name](const auto& name) { return name == property_name; });
+
+    if (optional_input != property_names.end()) {
+      result = optional_input - property_names.begin();
+    }
+
+    return result;
+  };
+
+  filename_input_ = find_optional_input("filename");
+  model_name_input_ = find_optional_input("model");
+
+  // OpenAI audio endpoints require 'file' and 'model'.
+  if (!std::any_of(property_names.begin(), property_names.end(),
+                   [](const auto& name) { return name == "file"; })) {
+    ORTX_CXX_API_THROW("Required 'file' input was not found", ORT_INVALID_ARGUMENT);
+  }
+
+  if (ModelName().empty() && !model_name_input_) {
+    ORTX_CXX_API_THROW("Required 'model' input was not found", ORT_INVALID_ARGUMENT);
+  }
+}
+
+void OpenAIAudioToTextInvoker::ValidateArgs(const ortc::Variadic& inputs) const {
+  // We don't have a way to get the output type from the custom op API.
+  // If there's a mismatch it will fail in the Compute when it allocates the output tensor.
+  if (OutputNames().size() != 1) {
+    ORTX_CXX_API_THROW("Expected single output", ORT_INVALID_ARGUMENT);
+  }
+}
+
+void OpenAIAudioToTextInvoker::SetupRequest(CurlHandler& curl_handler, const ortc::Variadic& inputs) const {
+  // theoretically the filename the content was buffered from. provides the extensions indicating the audio format
+  static const std::string fake_filename = "audio." + audio_format_;
+
+  const auto& property_names = PropertyNames();
+
+  const auto& get_optional_input =
```

Comment:
(For my own education) do we save anything by collecting the lambda in a reference here?

Reply:
no - shouldn't be a reference. typo.

Reply:
Typo needs correction.

Reply:
Fixed in branch with some other CI changes. Will be in next PR.